browser/android/preferences/background_video_playback_tab_helper.cc

```diff
@@ -39,14 +39,11 @@ bool IsBackgroundVideoPlaybackEnabled(content::WebContents* contents) {
     return false;
 
   const std::string host = contents->GetLastCommittedURL().host();
-  const std::string path = contents->GetLastCommittedURL().path();
-  if (path.find("/watch") != std::string::npos) {
-    if (host.find("www.youtube.com") != std::string::npos ||
-        host.find("youtube.com") != std::string::npos ||
-        host.find("m.youtube.com") != std::string::npos) {
-        content::RenderFrameHost::AllowInjectingJavaScript();
-        return true;
-    }
+  if (host.find("www.youtube.com") != std::string::npos ||
+      host.find("youtube.com") != std::string::npos ||
+      host.find("m.youtube.com") != std::string::npos) {
+      content::RenderFrameHost::AllowInjectingJavaScript();
```

Comment:
Will this cover music.youtube.com also?

Reply:
It should be, yes. As matter of fact we don't even need `www.youtube.com` and `m.youtube.com` as they are covered by `host.find("youtube.com")`

Reply:
Created follow up issue to remove excessive checks https://github.com/brave/brave-browser/issues/9548

Reply:
checks are not great because it can match things like youtube.com.evil.com
I think `net::registry_controlled_domains::SameDomainOrHost` should be used here

Reply:
Yep, will fix it in follow up issue